src/groups/circle_group.jl

```diff
@@ -90,3 +90,57 @@ _log_lie!(G::CircleGroup, X, q) = (X .= log_lie(G, q))
 function number_of_coordinates(G::CircleGroup, B::AbstractBasis)
     return number_of_coordinates(base_manifold(G), B)
 end
+
+@doc raw"""
+    RealCircleGroup <: GroupManifold{Circle{ℝ},AdditionOperation}
+
+The real circle group is the real circle ([`Circle(ℝ)`](@ref)) equipped with
+the group operation of addition ([`AdditionOperation`](@ref)).
+"""
+const RealCircleGroup = GroupManifold{ℝ,Circle{ℝ},AdditionOperation}
+
+RealCircleGroup() = GroupManifold(Circle{ℝ}(), AdditionOperation())
```

Comment:
Just as an idea – Could we also do `CircleGroup{F}`?

Reply:
I tried that, and the problem is that the complex circle group is multiplicative while real one is additive, and there is I think no other field that applies here, and we can't compute one type parameter based on other. So we would need two type parameters for `CircleGroup`, and the change would be breaking, so I thought introducing `RealCircleGroup` is a better solution.

Reply:
Ah ok.

Reply:
But you could do a conditional in the second argument of the GroupManifold? Or is that too strange?

Reply:
How would I do that?

Reply:
Sketch without testing (limited access atm):
```
CircleGroup(𝔽=ℝ) = GroulManifold(Circle{𝔽}(), 𝔽==ℝ ? AdditionOperation() : MultiplicationOperation())
```
or is this a problem with type stability or something?

Reply:
Yes, the constructor is not a problem, the type alias `CircleGroup` is (unless we don't care about it?)

Reply:
`CircleGroup` is exported though so we can't just discard the type alias part without making a breaking release. So the type alias has to stay for now, and it would be a bit weird IMO if `CircleGroup` constructor didn't return an instance of the `CircleGroup` type.

Reply:
then leave it as it is, i.e. to just cover the real-valued case.